enhancements/machine-config/on-cluster-layering.md

```diff
@@ -0,0 +1,1137 @@
+fairly straightforward: Since everything would exist in a tempdir within the
+build pod, deletion of the build pod would perform the cleanups. The
+downside of this approach is that should a failure occur, fetching the build
+inputs for inspection and debugging would be more difficult.
+
+Once the bootc ConfigMap annotation standards are better known, we could
+potentially skip the gzip / base64-encoding step by converting MachineConfigs
+into multiple ConfigMaps, temporarily storing them in the API server, injecting
+them into the build context, then deleting them once the build is complete.
+
+Within the build context itself, we'd need a different mechanism than Ignition
+live-apply to consume those ConfigMaps and inject their contents into the
+appropriate places within the image being built. Additionally, one of the big
+ideas around the bootc ConfigMap spec is using annotations to indicate what
+directory to store the files in, what file modes they should have, etc. This
+data would be unavailable via a volume mount. So it may be necessary to create a
+map of ConfigMap name -> bootc annotations and store this in an additional
+ConfigMap. This would also necessitate the need for a binary such as Ignition
+live-apply to consume this information from within the build context to write
+those files into the appropriate places.
+
+This could look like this instead:
+
+```mermaid
+flowchart TB
+  Start("Start")
+  subgraph Prepare [Prepare For Build]
+    direction TB
+    RenderContainerfile("Render Containerfile")
+    PrepMachineConfig("Convert into bootc ConfigMaps")
+    PrepIndexConfigMap("Prepare bootc index ConfigMap")
+    ConfigMaps("Create ephemeral ConfigMaps")
+  
+    RenderContainerfile --> ConfigMaps
+    PrepMachineConfig --> PrepIndexConfigMap --> ConfigMaps
+  end
+  
+  subgraph ImageBuilder [Image Builder Context]
+    direction TB
+    ParseIndexConfigMap("Parse bootc index ConfigMap")
+    ApplyConfigMapsToImage("Apply bootc ConfigMaps \n to image build")
+    RunAdminProvidedContent("Run admin-provided \n Containerfile content")
+    ParseIndexConfigMap-->ApplyConfigMapsToImage-->RunAdminProvidedContent
+  end
+
+  Stop("Stop")
+
+  Start --> Prepare
+  ConfigMaps --> |Mounted as volumes into image build context| ImageBuilder
+  IsSuccess{"Is Build Successful?"}
+
+  ImageBuilder --> IsSuccess
+  IsSuccess --> | Yes | DeleteConfigMaps("Delete ConfigMaps")
+  IsSuccess --> | No | LeaveConfigMaps("Leave ConfigMaps \n for inspection")
+  DeleteConfigMaps --> Stop
+  LeaveConfigMaps --> Stop
+```
+
+To be clear, these ConfigMaps would be created and managed by the
+BuildController process. A cluster admin would never be expected to create or
+maintain these. This may also not be required if we were to adopt the [Custom OS
+Content Only](#####Custom-OS-Content-Only) idea.
+
+#### Build Pods
+
+Because we cannot rely upon OpenShift Image Builds to be present, there is
```

Comment:
Could you make Image Builds a requirement of the feature to avoid this complexity? Force users to turn it on, to be able to use this feature?